rust/processor/src/db/common/models/default_models/parquet_move_resources.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+#![allow(clippy::extra_unused_lifetimes)]
+
+use crate::{
+    parquet_processors::generic_parquet_processor::{HasVersion, NamedTable},
+    utils::util::standardize_address,
+};
+use allocative_derive::Allocative;
+use anyhow::{Context, Result};
+use aptos_protos::transaction::v1::{
+    DeleteResource, MoveStructTag as MoveStructTagPB, WriteResource,
+};
+use field_count::FieldCount;
+use parquet_derive::ParquetRecordWriter;
+use serde::{Deserialize, Serialize};
+
+#[derive(
+    Allocative, Clone, Debug, Default, Deserialize, FieldCount, Serialize, ParquetRecordWriter,
+)]
+pub struct MoveResource {
+    pub txn_version: i64,
+    pub write_set_change_index: i64,
+    pub block_height: i64,
+    #[allocative(skip)]
+    pub block_timestamp: chrono::NaiveDateTime,
+    pub resource_address: String,
+    pub resource_type: String,
+    pub module: String,
+    pub fun: String,
+    pub is_deleted: bool,
+    pub generic_type_params: Option<String>,
+    pub data: Option<String>,
+    pub state_key_hash: String,
+}
+
+impl NamedTable for MoveResource {
+    const TABLE_NAME: &'static str = "move_resources";
+}
+
+impl HasVersion for MoveResource {
+    fn version(&self) -> i64 {
+        self.txn_version
+    }
+}
+
+pub struct MoveStructTag {
+    resource_address: String,
+    pub module: String,
+    pub fun: String,
+    pub generic_type_params: Option<String>,
+}
+
+impl MoveResource {
+    pub fn from_write_resource(
+        write_resource: &WriteResource,
+        write_set_change_index: i64,
+        txn_version: i64,
+        block_height: i64,
+        block_timestamp: chrono::NaiveDateTime,
+    ) -> Self {
+        let parsed_data = Self::convert_move_struct_tag(
+            write_resource
+                .r#type
+                .as_ref()
+                .expect("MoveStructTag Not Exists."),
+        );
+        Self {
+            txn_version,
+            block_height,
+            write_set_change_index,
+            resource_type: write_resource.type_str.clone(),
+            fun: parsed_data.fun.clone(),
+            resource_address: standardize_address(&write_resource.address.to_string()),
+            module: parsed_data.module.clone(),
+            generic_type_params: parsed_data.generic_type_params,
+            data: Some(write_resource.data.clone()),
+            is_deleted: false,
+            state_key_hash: standardize_address(
+                hex::encode(write_resource.state_key_hash.as_slice()).as_str(),
+            ),
+            block_timestamp,
+        }
+    }
+
+    pub fn from_delete_resource(
+        delete_resource: &DeleteResource,
+        write_set_change_index: i64,
+        txn_version: i64,
+        block_height: i64,
+        block_timestamp: chrono::NaiveDateTime,
+    ) -> Self {
```

Comment:
) -> Self {
        let parsed_data = Self::convert_move_struct_tag(
            delete_resource
                .r#type
                .as_ref()
                .expect("MoveStructTag Not Exists."),
        );